src/backend/index.js

```diff
@@ -5,16 +5,22 @@ import { init } from './src/app.js'
 /** @type {import('util').ParseArgsConfig['options']} */
 const options = {
   version: { type: 'string' },
+  rootKey: { type: 'string' },
 }
 
 try {
   const args = parseArgs({ options })
 
   const { values } = args
 
+  if (typeof values.rootKey !== 'string') {
+    throw new Error('backend did not recieve root key from front end')
+  }
```

Comment:
don't think this check is needed because of the `parseArgs()` call, which strictly checks the flags passed and errors if it doesn't match the expectations. 

```suggestion
  if (typeof values.rootKey !== 'string') {
    throw new Error('backend did not recieve root key from front end')
  }
```

Reply:
this also checks for type error. `rootkey` can technically be a boolean as well

Reply:
the options configuration is set to make sure that we pass a string I think

https://nodejs.org/docs/latest-v16.x/api/util.html#utilparseargsconfig

Reply:
yeah, its typescript that's complaining though. The other option would be to use `ts-expect-error`, but this throw, while unnecessary, does feel a little clearer

Reply:
ah got it! thanks for clarifying